### ISSUE TITLE
Failure during recovery test

### ISSUE THREAD

**Issue by frelon** [rancher/elemental-toolkit#2076]
Seems to have started after merging SELinux support.

Logs:

```
      INFO[2024-04-30T15:52:15Z] Applying 'post-upgrade' hook                 
      INFO[2024-04-30T15:52:15Z] Running post-upgrade hook                    
      DEBU[2024-04-30T15:52:15Z] Running cmd: 'snapper --no-dbus --root /run/elemental/state/.snapshots/1/snapshot --csvout list --columns number,default,active' 
      DEBU[2024-04-30T15:52:15Z] 'snapper' command reported an error: exit status 1 
      DEBU[2024-04-30T15:52:15Z] 'snapper' command output: fsetfilecon on /run/elemental/state/.snapshots/1/snapshot/.snapshots failed errno: 30 (Read-only file system)
      IO error (.snapshots is not a btrfs subvolume). 
      ERRO[2024-04-30T15:52:15Z] failed collecting current snapshots: fsetfilecon on /run/elemental/state/.snapshots/1/snapshot/.snapshots failed errno: 30 (Read-only file system)
      IO error (.snapshots is not a btrfs subvolume). 
      ERRO[2024-04-30T15:52:15Z] failed getting snapshots list                
      ERRO[2024-04-30T15:52:15Z] failed upgrading installation metadata       
      INFO[2024-04-30T15:52:15Z] Deleting snapshot 2                          
      DEBU[2024-04-30T15:52:15Z] Running cmd: 'snapper --no-dbus --root /run/elemental/state/.snapshots/1/snapshot --csvout list --columns number,default,active' 
      DEBU[2024-04-30T15:52:15Z] 'snapper' command reported an error: exit status 1 
      DEBU[2024-04-30T15:52:15Z] 'snapper' command output: fsetfilecon on /run/elemental/state/.snapshots/1/snapshot/.snapshots failed errno: 30 (Read-only file system)
      IO error (.snapshots is not a btrfs subvolume). 
      ERRO[2024-04-30T15:52:15Z] failed collecting current snapshots: fsetfilecon on /run/elemental/state/.snapshots/1/snapshot/.snapshots failed errno: 30 (Read-only file system)
      IO error (.snapshots is not a btrfs subvolume). 
      ERRO[2024-04-30T15:52:15Z] failed listing available snapshots: exit status 1 
      DEBU[2024-04-30T15:52:15Z] Unmounting partition COS_PERSISTENT          
      DEBU[2024-04-30T15:52:16Z] Unmounting partition COS_STATE               
      DEBU[2024-04-30T15:52:16Z] Mounting partition COS_GRUB            
```

**Comment by frelon** [comment]
Still fails, but new place:

```
       INFO[2024-05-27T07:11:26Z] Applying 'post-upgrade' hook                 
      INFO[2024-05-27T07:11:26Z] Running post-upgrade hook                    
      DEBU[2024-05-27T07:11:26Z] Running cmd: 'snapper --no-dbus --root /run/elemental/state/.snapshots/1/snapshot --csvout list --columns number,default,active' 
      DEBU[2024-05-27T07:11:26Z] 'snapper' command reported an error: exit status 1 
      DEBU[2024-05-27T07:11:26Z] 'snapper' command output: IO error (.snapshots is not a btrfs subvolume). 
      ERRO[2024-05-27T07:11:26Z] failed collecting current snapshots: IO error (.snapshots is not a btrfs subvolume). 
      ERRO[2024-05-27T07:11:26Z] failed getting snapshots list                
      ERRO[2024-05-27T07:11:26Z] failed upgrading installation metadata       
      INFO[2024-05-27T07:11:26Z] Deleting snapshot 2                          
      DEBU[2024-05-27T07:11:26Z] Running cmd: 'snapper --no-dbus --root /run/elemental/state/.snapshots/1/snapshot --csvout list --columns number,default,active' 
      DEBU[2024-05-27T07:11:26Z] 'snapper' command reported an error: exit status 1 
      DEBU[2024-05-27T07:11:26Z] 'snapper' command output: IO error (.snapshots is not a btrfs subvolume). 
      ERRO[2024-05-27T07:11:26Z] failed collecting current snapshots: IO error (.snapshots is not a btrfs subvolume). 
      ERRO[2024-05-27T07:11:26Z] failed listing available snapshots: exit status 1 
      DEBU[2024-05-27T07:11:26Z] Unmounting partition COS_PERSISTENT          
```

**Comment by davidcassany** [comment]
Fixed on #2098
